src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -2492,8 +2492,8 @@ private void ClassifyUpdate(VariableDeclaratorSyntax oldNode, VariableDeclarator
                 }
 
                 // Check if a constant field is updated:
-                var fieldDeclaration = (FieldDeclarationSyntax)oldNode.Parent.Parent;
-                if (fieldDeclaration.Modifiers.Any(SyntaxKind.ConstKeyword))
+                if ((oldNode.Parent.Parent is FieldDeclarationSyntax fieldDeclaration) &&
```

Comment:
I don't quite like this. We already know we are dealing with a [variable declaration](https://github.com/dotnet/roslyn/pull/35962/files#diff-6e3a4030aa2c63674655cf9ec604543cR2468) within [a type declaration](https://github.com/dotnet/roslyn/pull/35962/files#diff-6e3a4030aa2c63674655cf9ec604543cR2487).

That means it has to be a field declaration of some kind. Therefore the check is unnecessary.

Reply:
@tmat ,  the crash (#32889) we are solving is the variable declaration with the type declaration where the variable is not a _field_ but an _event field_. I think that the unit test provided covers this. Not?

Reply:
@tmat please review my previous comment.

Reply:
@tmat please review

Reply:
@tmat please review

Reply:
I prefer the original fix, that is cast to `BaseFieldDeclarationSyntax`. In this case `const` is not applicable to event fields, but that's fine. Then it won't be found in the modifiers anyways.

Reply:
@tmat I have reverted. Please sign off.
@jinujoseph @ManishJayaswal please consider for 16.2 Preview 4 / GA. It seems to be a pretty easy fix.